compiler/codegen/CodeGenRA.cpp

```diff
@@ -452,8 +452,8 @@ OMR::CodeGenerator::allocateSpill(bool containsCollectedReference, int32_t *offs
 TR_BackingStore *
 OMR::CodeGenerator::allocateSpill(int32_t dataSize, bool containsCollectedReference, int32_t *offset, bool reuse)
    {
-   TR_ASSERT(dataSize <= 16, "assertion failure");
-   TR_ASSERT(!containsCollectedReference || (dataSize == TR::Compiler->om.sizeofReferenceAddress()), "assertion failure");
+   TR_ASSERT_FATAL(dataSize <= 16, "assertion failure");
+   TR_ASSERT_FATAL(!containsCollectedReference || (dataSize == TR::Compiler->om.sizeofReferenceAddress()), "assertion failure");
```

Comment:
A lot of the assertions in RA have been made fatal to catch problems at compile time rather than exhibit weird runtime crashes. The cost for these assertions is so small that it is definitely worth while to enable them.